File: GolfTournamentData/Models/Team.cs
// GolfTournamentData/Models/Team.cs
using System.ComponentModel.DataAnnotations;

namespace GolfTournamentData.Models
{
    public class Team
    {
        public int TeamId { get; set; }
        public int TournamentId { get; set; }

        [Required]
        [StringLength(100)]
        public string TeamName { get; set; } = string.Empty;

        [Required]
        [StringLength(100)]
        public string Player1Name { get; set; } = string.Empty;

        [Required]
        [StringLength(100)]
        public string Player2Name { get; set; } = string.Empty;

        // Navigation properties
        public Tournament Tournament { get; set; } = null!;
        public ICollection<Score> Scores { get; set; } = new List<Score>();
    }
}

File: golf-tournament-frontend/src/types/team.ts
// src/types/team.ts
export interface Team {
  teamId: number;
  name: string;
  teamName: string; // For backwards compatibility
  tournamentId: number;
  player1Name: string;
  player2Name: string;
}

export interface CreateTeamRequest {
  name: string;
  teamName: string;
  tournamentId: number;
  player1Name: string;
  player2Name: string;
}